src/components/MonitorList.vue

```diff
@@ -81,31 +81,71 @@ export default {
 
         },
 
+        /**
+         * Returns a sorted list of monitors based on the applied filters and search text.
+         *
+         * @return {Array} The sorted list of monitors.
+         */
         sortedMonitorList() {
             let result = Object.values(this.$root.monitorList);
 
-            // Simple filter by search text
-            // finds monitor name, tag name or tag value
-            if (this.searchText !== "") {
-                const loweredSearchText = this.searchText.toLowerCase();
-                result = result.filter(monitor => {
-                    return monitor.name.toLowerCase().includes(loweredSearchText)
+            result = result.filter(monitor => {
+
+                // filter by search text
+                // finds monitor name, tag name or tag value
+                let searchTextMatch = true;
+                if (this.searchText !== "") {
+                    const loweredSearchText = this.searchText.toLowerCase();
+                    searchTextMatch =
+                        monitor.name.toLowerCase().includes(loweredSearchText)
                         || monitor.tags.find(tag => tag.name.toLowerCase().includes(loweredSearchText)
                             || tag.value?.toLowerCase().includes(loweredSearchText));
-                });
-            } else {
-                result = result.filter(monitor => monitor.parent === null);
-            }
+
+                }
+
+                // filter by status
+                let statusMatch = true;
+                if (this.filterState.status != null && this.filterState.status.length > 0) {
+                    if (monitor.id in this.$root.lastHeartbeatList && this.$root.lastHeartbeatList[monitor.id]) {
+                        monitor.status = this.$root.lastHeartbeatList[monitor.id].status;
+                    }
+                    statusMatch = this.filterState.status.includes(monitor.status);
+                }
+
+                // filter by active
+                let activeMatch = true;
+                if (this.filterState.active != null && this.filterState.active.length > 0) {
+                    activeMatch = this.filterState.active.includes(monitor.active);
+                }
```

Comment:
You seem to like the "if.. assign" pattern (don't know the real name)

I would argue that this can be simplified to
```suggestion
                const shouldFilterByActive = this.filterState.active != null;
                let activeMatch = shouldFilterByActive && this.filterState.active.includes(monitor.active);
```

(code-style => feel free to overrule ^^)

Reply:
I have updated the proposed code (this time with a boolean table):
|               | `filterCanBeActive` true | `filterCanBeActive` false |
|---------------|---------|----------|
| `includes` true  | true    | true     |
| `includes` false | false   | true     |

Reply:
I guess I see that this makes it a bit shorter, but I don't think this makes it more readable, which is important to me.

Reply:
True, the `!filterCanBeActive` is hideous.
I proposed this to be explicit about not changing any logic this time (introducing bugs is worse than hard to maintain code) ^^

I think it is more readable if `!filterCanBeActive` is integrated to `filterCannotBeActive` and the expression changed in the corresponding way.

As stated above, if you feel differently, just overrule my comment and close this review-comment